frontend/vue/components/CodeExercise/ExerciseActionsBar.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div class="exercise-actions-bar">
+    <button
+      v-if="runEnabled && !isRunning"
+      class="exercise-actions-bar__button exercise-actions-bar__button_run"
+      @click="run()"
+    >
+      {{ $translate('Run') }}
+    </button>
+    <button
+      v-if="gradeEnabled && !isRunning"
+      class="exercise-actions-bar__button"
+      @click="grade()"
+    >
+      {{ $translate('Grade') }}
+    </button>
+    <div v-if="isRunning" class="exercise-actions-bar__running-indicator">
+      <bx-loading class="exercise-actions-bar__running-indicator__icon" assistive-text="Running" type="small" />
+      <span class="exercise-actions-bar__running-indicator__label">{{ $translate('Running') }}</span>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue-demi'
+import 'carbon-web-components/es/components/loading/loading'
+
+export default defineComponent({
+  name: 'ExerciseActionsBar',
+  props: {
+    isRunning: {
+      type: Boolean,
+      required: false,
+      default: true
+    },
+    runEnabled: {
+      type: Boolean,
+      required: false,
+      default: true
+    },
+    gradeEnabled: {
+      type: Boolean,
+      required: false,
+      default: true
+    }
+  },
+  methods: {
+    run () {
+      if (this.isRunning) {
+        return
+      }
+      this.$emit('run')
```

Comment:
We can simplify this:

```suggestion
      if (!this.isRunning) {
        this.$emit('run')
      }
```